src/config/ability.js

```diff
@@ -0,0 +1,25 @@
+import { AbilityBuilder } from '@casl/ability'
+
+export default function defineAbilitiesFor (ctx) {
+  let user = (ctx.state.isAuthenticated) ? ctx.state.user : {roles: ['public']}
+  let roleExists = (roleArray, roleName) => {
```

Comment:
Normally I would move the function to the outer scope. Keeps this function here cleaner.

On another note, we should use [Array.prototype.includes()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes). And since this is a oneliner we can kill the function after all.